### PR TITLE
Support explicitly enabling hostname verificaiton

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
OpenSSL doesn't automatically verify hostnames and requires extract method calls to enable this feature [1]. We should allow this to be configured.

Modifications:
- Enabled calling of X509_VERIFY_PARAM_set_hostflags and X509_VERIFY_PARAM_set1_host

Result:
Support for hostname verification.

[1] https://wiki.openssl.org/index.php/Hostname_validation

### CODE REVIEW

**Comment by normanmaurer** [comment]
@Scottmitch this looks great I just wonder if we not need to take the algorithm into account. See https://github.com/netty/netty/pull/6403#discussion_r101694705 .

File: openssl-dynamic/src/main/c/ssl.c [hunk 1]
```diff
@@ -1979,6 +2011,8 @@ TCN_IMPLEMENT_CALL(void, SSL, setVerify)(TCN_STDARGS, jlong ssl, jint level, jin
            tcn_ThrowException(e, "failed to allocate tcn_ssl_verify_config_t");
            return;
        }
+       // Copy the verify depth form the context in case depth is <0.
+       verify_config->verify_depth = c->verify_config.verify_depth;
```
Comment (normanmaurer) [inline openssl-dynamic/src/main/c/ssl.c]: @Scottmitch should this be a different PR ? Seems like its not really related.

File: openssl-dynamic/src/main/c/sslutils.c [hunk 2]
```diff
@@ -571,7 +571,9 @@ int tcn_X509_up_ref(X509* cert) {
 }
 
 int tcn_set_verify_config(tcn_ssl_verify_config_t* c, jint tcn_mode, jint depth) {
-    c->verify_depth = depth > 0 ? depth : 0;
+    if (depth >= 0) {
+        c->verify_depth = depth;
+    }
```
Comment (normanmaurer) [inline openssl-dynamic/src/main/c/sslutils.c]: also seems not related to this PR

File: openssl-dynamic/src/main/c/ssl.c (hunk 1 above)
Comment (Scottmitch) [inline openssl-dynamic/src/main/c/ssl.c]: This PR creates the need to call `setVerify` and ignore the `depth`. Before this PR the depth would always be set so it didn't matter.

Comment (normanmaurer) [inline openssl-dynamic/src/main/c/ssl.c]: ah got it... thanks 👍

File: openssl-dynamic/src/main/c/sslutils.c (hunk 2 above)
Comment (Scottmitch) [inline openssl-dynamic/src/main/c/sslutils.c]: This PR creates the need to call `setVerify` and ignore the `depth` ... a change in behavior that wasn't necessary/supported before.

**Comment by Scottmitch** [comment]
master (df8f3b1)
